packages/medusa/src/services/fulfillment-provider.ts

```diff
@@ -0,0 +1,186 @@
+import { MedusaError } from "medusa-core-utils"
+import BaseFulfillmentService from "medusa-interfaces/dist/fulfillment-service"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import {
+  Cart,
+  Fulfillment,
+  FulfillmentProvider,
+  LineItem,
+  Order,
+  Return,
+  ShippingMethod,
+  ShippingOption,
+} from "../models"
+import { FulfillmentProviderRepository } from "../repositories/fulfillment-provider"
+import { CreateFulfillmentOrder } from "../types/fulfillment"
+import { FulfillmentOptions } from "../types/fulfillment-provider"
+import { MedusaContainer } from "../types/global"
+
+type NotificationProviderKey = `fp_${string}`
+
+type FulfillmentProviderContainer = MedusaContainer & {
+  fulfillmentProviderRepository: typeof FulfillmentProviderRepository
+  manager: EntityManager
+} & {
+  [key in `${NotificationProviderKey}`]: BaseFulfillmentService
+}
+
+/**
+ * Helps retrive fulfillment providers
+ */
+class FulfillmentProviderService extends TransactionBaseService<FulfillmentProviderService> {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly container_: FulfillmentProviderContainer
+
+  protected readonly fulfillmentProviderRepository_: typeof FulfillmentProviderRepository
+
+  constructor(container: FulfillmentProviderContainer) {
+    super(container)
+
+    const { manager, fulfillmentProviderRepository } = container
+
+    this.container_ = container
+    this.manager_ = manager
+    this.fulfillmentProviderRepository_ = fulfillmentProviderRepository
+  }
+
+  async registerInstalledProviders(providers: string[]): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const fulfillmentProviderRepo = manager.getCustomRepository(
+        this.fulfillmentProviderRepository_
+      )
+      await fulfillmentProviderRepo.update({}, { is_installed: false })
+
+      for (const p of providers) {
+        const n = fulfillmentProviderRepo.create({ id: p, is_installed: true })
+        await fulfillmentProviderRepo.save(n)
+      }
+    })
+  }
+
+  async list(): Promise<FulfillmentProvider[]> {
+    const fpRepo = this.manager_.getCustomRepository(
+      this.fulfillmentProviderRepository_
+    )
+
+    return await fpRepo.find({})
+  }
+
+  async listFulfillmentOptions(
+    provider_ids: string[]
```

Comment:
todo: Should be camelCased
```suggestion
    providerIds: string[]
```